client/aggregator.go

```diff
@@ -24,32 +30,38 @@ type subscriber struct {
 
 type watchAggregator struct {
 	Client
-	log log.Logger
+	autoStart bool
+	log       log.Logger
 
 	subscriberLock sync.Mutex
 	subscribers    []subscriber
 }
 
+// SetLog configures the client log output
+func (c *watchAggregator) SetLog(l log.Logger) {
+	c.log = l
+}
+
 func (c *watchAggregator) Watch(ctx context.Context) <-chan Result {
 	c.subscriberLock.Lock()
 	defer c.subscriberLock.Unlock()
 
 	sub := subscriber{ctx, make(chan Result, 5)}
 	c.subscribers = append(c.subscribers, sub)
 
-	if len(c.subscribers) == 1 {
+	if len(c.subscribers) == 1 && !c.autoStart {
 		ctx, cancel := context.WithCancel(context.Background())
-		go c.distribute(c.Client.Watch(ctx), cancel)
+		go c.distribute(c.Client.Watch(ctx), cancel, false)
 	}
 	return sub.c
 }
 
-func (c *watchAggregator) distribute(in <-chan Result, cancel context.CancelFunc) {
+func (c *watchAggregator) distribute(in <-chan Result, cancel context.CancelFunc, keepAlive bool) {
 	defer cancel()
 	for {
 		var aCtx context.Context
 		c.subscriberLock.Lock()
-		if len(c.subscribers) == 0 {
+		if len(c.subscribers) == 0 && !keepAlive {
```

Comment:
If `keepAlive` is `true` and there's no subscribers (like when this is called in the constructor) does the line below cause a nil pointer: `aCtx = c.subscribers[0].ctx`? Needs a test(s)?